### PR TITLE
feat: `boltup` installer + script

### PR DESCRIPTION
**Pull request by merklefruit**
Quick installer for the `bolt` CLI tool.
Mostly copied from the amazing [foundryup](https://book.getfoundry.sh/getting-started/installation#using-foundryup) 👑.

> [!Note]
> the install script relies on a binary existing on the unstable branch, so this won't work until this PR is merged.

Try it with:

```sh
curl -L https://raw.githubusercontent.com/chainbound/bolt/boltup/boltup/install.sh | bash
```

I am not responsible if all your files are deleted, etc etc

### CODE REVIEW

**Comment by mempirate** [comment]
Nice, we'll host this on `https://boltprotocol.xyz/install.sh`

File: boltup/boltup.sh [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        exit 0
+        ;;
+      *)
+        warn "unknown option: $1"
+        usage
+        exit 1
+    esac; shift
+  done
+
+  CARGO_BUILD_ARGS=(--release)
```
Comment (mempirate) [inline boltup/boltup.sh]: This is unused right?

File: boltup/boltup.sh [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+# 'boltup' script for installing and managing the bolt CLI tool.
+#
+# This script is a heavily inspired by 'foundryup' (https://github.com/foundry-rs/foundry/blob/master/foundryup/foundryup)
+# since it is a great way to install CLI tools. Kudos to the Foundry team :)
+
+BASE_DIR=${XDG_CONFIG_HOME:-$HOME}
+BOLT_DIR=${BOLT_DIR:-"$BASE_DIR/.bolt"}
+BOLT_BIN_DIR="$BOLT_DIR/bin"
+
+export RUSTFLAGS="${RUSTFLAGS:--C target-cpu=native}"
```
Comment (mempirate) [inline boltup/boltup.sh]: Not used

**Review by mempirate — CHANGES_REQUESTED** [review]
Nice, smol nits

File: boltup/boltup.sh [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      -h|--help)
+        usage
+        exit 0
+        ;;
+      *)
+        warn "unknown option: $1"
+        usage
+        exit 1
+    esac; shift
+  done
```
Comment (thedevbirb) [inline boltup/boltup.sh:36]: Can this be moved to a separate, documented function?
By documentation I mean something like:

"This function loops as long as $1 (the first positional argument) is not empty, and matches its value against flags. If a flag is found, the argument is discarded (via `shift`) and the value for the flag is read."

I know, I should RTFM but having gently reminders of bash syntax is nice.

**Review by thedevbirb — APPROVED** [review]
wen nix

File: boltup/boltup.sh (hunk 3 above)
Comment (merklefruit) [reply boltup/boltup.sh:36]: I didn't move it but I added the comment, as I don't want to introduce untested behaviour in a script that is battle tested (this is what foundry has been using for 2+ years)

**Pull request merged**